src/components/Dialog/index.tsx

```diff
@@ -105,19 +104,34 @@ export function Outer({
     [setOpenIndex, setDialogIsOpen, control.id],
   )
 
+  // This is the function that we call when we want to dismiss the dialog.
   const close = React.useCallback<DialogControlProps['close']>(cb => {
-    if (cb && typeof cb === 'function') {
-      if (closeCallback.current) {
-        logger.error(
-          `Dialog close was passed multiple callbacks, you shouldn't do that`,
-        )
-      }
+    if (typeof cb === 'function') {
       closeCallback.current = cb
     }
-    // initiates a close animation, the actual "close" happens in `onCloseInner`
     sheet.current?.close()
   }, [])
 
+  // This is the actual thing we are doing once we "confirm" the dialog. We want the dialog's close animation to
+  // happen before we run this. It is passed to the `BottomSheet` component.
+  const onCloseAnimationComplete = React.useCallback(() => {
```

Comment:
Calling this `onCloseAnimationComplete` I think gives us a better sense of the purpose of this function. Too many `onClose`s gets a bit confusing I think.